src/submission/validation-clinical/specimen.ts

```diff
@@ -67,18 +67,22 @@ function getSpecimenFromDonor(
   specimenRecord: DeepReadonly<SubmittedClinicalRecord>,
   validationResults: RecordValidationResult[],
 ) {
-  const specimen = _.find(existentDonor.specimens, [
-    'submitterId',
-    specimenRecord[SampleRegistrationFieldsEnum.submitter_specimen_id],
-  ]);
+  const specimen = getSingleClinicalObjectFromDonor(
+    existentDonor,
+    ClinicalEntitySchemaNames.SPECIMEN,
+    {
+      submitterId: specimenRecord[SpecimenFieldsEnum.submitter_specimen_id],
```

Comment:
just for consistency I prefer we use `ClinicalUniqueIndentifier` wherever we want to get the id field by schema .. there is a very low chance of say we change from `drug name` to another `chemo_therapy_submitter_id` for example

Reply:
I'll mark the locations that I found this in this PR